### DISCUSSION TITLE
Feature request: Delete trashed notes after 30 days

### DISCUSSION THREAD

**Discussion by pienkowb** [standardnotes/forum#2790] [Suggestions]
It would be great if the application could automatically delete a note that was moved to trash after some period of time (e.g. 30 days), similarly to how Apple Notes does it.

This of course should be an opt-in feature.

**Comment by atmoio** [comment]
I don't like the idea too much of having a script that automatically permanently deletes notes. Best to keep this to the user, similar to how operating system trash/recycle bins work.

**Comment by pienkowb** [comment]
I don't know about other operating systems but macOS has this functionality built in (see [this](https://itectec.com/askdifferent/macos-permanently-deleting-trashed-files-after-30-days/
)).

I would argue it's a very useful feature for people who care about trashed notes just for some period of time and don't want to delete the old ones manually. In case an important note is deleted too soon, everything is backed up anyway.

**Comment by atmoio** [comment]
With 30-day autodelete, what if a file is trashed on Day 1, but you don't open the app again until Day 40? It would only then delete on launch. So the communication behind a theoretical feature like this would have to make clear it's not the server that does the auto-deletion, but an unlocked and authenticated application.

**Comment by pienkowb** [comment]
Yes, that makes sense.

**Comment by wenman-ualberta** [comment]
> With 30-day autodelete, what if a file is trashed on Day 1, but you don't open the app again until Day 40? It would only then delete on launch. So the communication behind a theoretical feature like this would have to make clear it's not the server that does the auto-deletion, but an unlocked and authenticated application.

I think having the application do it automatically is still a bad idea, and is furthermore against Standard Notes' commitment to longevity. I also think it's a bad design flow for destructive processes to not require human input.

@pienkowb What do you think about this idea:?

There's an option for a 30-day *prompt* for you to delete notes that are that old (or older) in the trash can when you sign in?

The ultimate UI situation would be for the confirmation window to include a list of the notes being deleted, with their visibility matching their previous status. So a "locked' note would still require a password prompt if you wanted to look at what it was before you delete it. Otherwise the 'Do you want to delete these files?" prompt would be an attack vector for reading notes.

This nice thing about a prompt with a brief preview is that it strikes the perfect balance, in my opinion, of convince and control. It's convenient in that you don't have to remember to delete your files, it does that for you; yet you are always in control. If at the last second you release that you don't want to delete that particular file that you put in the trash earlier, the prompt acts as a final confirmation that you truly, absolutely want to get rid of it. You're always in control.

What do you think?

**Comment by pienkowb** [comment]
> There's an option for a 30-day _prompt_ for you to delete notes that are that old (or older) in the trash can when you sign in?
> 
> The ultimate UI situation would be for the confirmation window to include a list of the notes being deleted, with their visibility matching their previous status. So a "locked' note would still require a password prompt if you wanted to look at what it was before you delete it. Otherwise the 'Do you want to delete these files?" prompt would be an attack vector for reading notes.
> 
> This nice thing about a prompt with a brief preview is that it strikes the perfect balance, in my opinion, of convince and control. It's convenient in that you don't have to remember to delete your files, it does that for you; yet you are always in control. If at the last second you release that you don't want to delete that particular file that you put in the trash earlier, the prompt acts as a final confirmation that you truly, absolutely want to get rid of it. You're always in control.
> 
> What do you think?

@A-Casimir-Wenman I really like this idea. There's just one issue – a user who moves notes to trash on a daily basis will be prompted to confirm deletion almost daily (because every day there will be a note that reached the 30-day period).

**Comment by fabien-joubert** [comment]
I would love to see this feature come out the way Apple implemented it: just an opt-in. No need for user confirmation to automatically delete notes: opt-in is the confirmation.
To go further, we should be able to select the retention period for notes in the trash from 7 to 365 days.

However, I think that this feature may break the first principle of the Longevity Statement by adding more complexity than needed to the software. 
After all, the most basic feature of the trash is to store the notes you want to delete. And then be able to permanently delete or recover the notes. 
In that regard everything needed to have a good trash is already implemented: the user can already delete one by one or all the notes from the trash. 

Multi-select feature could improve the experience of deleting notes. But adding additional features like automatic trash deletion is a trade-off between adding features and keeping simplicity.

Do we want Standard Notes to automatically delete our notes in the trash or not? 
What do we, as users, gain from this behavior?

I think we can gain peace of mind knowing that we can forget a note in the trash and it will be removed automatically (again, only if we asked to do so). Even some passwords managers do that (Bitwarden: 30 days).
